lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1598,9 +1598,19 @@ protected function _validateAndCompleteOneToOneMapping(array $mapping)
                 if ( ! $this->table) {
                     throw new RuntimeException("ClassMetadataInfo::setTable() has to be called before defining a one to one relationship.");
                 }
+                /*
+                 * Mapping via the field name causes issues if unrelated tables use the same fieldName in their independent mappings
+                 * Note: The hack below of prepending the table name probably exceeds length constraints of some databases
+                 * e.g. Oracle.
+                 * Naming should really be delegated to the NamingStrategy but there is no API in the interface to support this yet.
+
                 $this->table['uniqueConstraints'][$mapping['fieldName']."_uniq"] = array(
                     'columns' => $uniqueConstraintColumns
                 );
+                 */
+                $this->table['uniqueConstraints'][$this->table['name'].$mapping['fieldName']."_uniq"] = array(
```

Comment:
Yes, hence all my questions on the google groups.

To avoid breaking compatibility I would be proposing extending the Naming Interface and leaving the existing behaviour. For those people who need this fix they can change it easily via a new naming interface instance.

Can I get some agreement that this is a real bug?